editor/src/components/canvas/ui-jsx-canvas-renderer/ui-jsx-canvas-component-renderer.tsx

```diff
@@ -341,20 +341,29 @@ export function createComponentRendererComponent(params: {
   return Component
 }
 
+function isRenderProp(prop: any): prop is { [UTOPIA_PATH_KEY]: string; props: MapLike<any> } {
+  return ((prop as any)?.props as any)?.[UTOPIA_PATH_KEY] != null
```

Comment:
hmmmm 😅 
could this be made a little less scary/more readable?

Reply:
refactored in https://github.com/concrete-utopia/utopia/pull/5559/commits/79fe89e44328f97c545ecf6d7322258366d8363d